### ISSUE TITLE
Swiper: `Error in post-command-hook (ivy--queue-exhibit): (wrong-type-argument number-or-marker-p nil)` (with org-mode?)

### ISSUE THREAD

**Issue by stefan2904** [abo-abo/swiper#3036]
Since a few weeks, I get the following error when using swiper (after a few keystrokes in the swiper minibuffer):

`Error in post-command-hook (ivy--queue-exhibit): (wrong-type-argument number-or-marker-p nil)`

It seems that this only happens in buffers with `org-mode` enabled, but I am not 100% sure. 
As this does not trigger debug-on-error, I am not sure how to best debug it further. 

My init.el only contains the following:

```elisp

(use-package ivy
  :ensure t)
(use-package swiper
  :ensure t)

(require 'ivy)

;(ivy-mode 1)

(setq ivy-use-virtual-buffers t)
;(setq ivy-count-format "(%d/%d) ")
;(setq enable-recursive-minibuffers t)


(global-set-key "\C-s" 'swiper)
```

- GNU Emacs 28.1
- counsel                        20240229.1509
- ivy                            20240214.2118
- swiper                         20240214.2118
- Org mode version 9.7-pre (release_9.6.21-1298-gb338a9)

**Comment by basil-conto** [comment]
> `Error in post-command-hook (ivy--queue-exhibit): (wrong-type-argument number-or-marker-p nil)`
> 
> It seems that this only happens in buffers with `org-mode` enabled, but I am not 100% sure. As this does not trigger debug-on-error, I am not sure how to best debug it further.

Yes, these can be difficult to debug, especially when it's not always reproducible.  One of the simplest options is to copy the definition of `ivy--insert-minibuffer`, and sprinkle some print-debugging calls like `(message ">>> n")` for different values of `n` among the function's lines.  That might shed more light on which part of the `ivy--exhibit` hook is failing.  It could be that `old-mark` is `nil`, for example.

**Comment by basil-conto** [comment]
Ping.  Are you still seeing this issue?
